src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/ProcessExecutor.cs

```diff
@@ -0,0 +1,206 @@
+                        }
+                    }
+                };
+
+                bool processStarted = false;
+                try
+                {
+                    processStarted = process.Start();
+                    if (!processStarted)
+                    {
+                        Logger.LogError("Failed to start process: {Command}", command);
+                        return (false, string.Empty, "Failed to start process");
+                    }
+
+                    process.BeginOutputReadLine();
+                    process.BeginErrorReadLine();
+
+                    if (timeout.HasValue)
+                    {
+                        using CancellationTokenSource timeoutCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                        timeoutCts.CancelAfter(timeout.Value);
+
+                        try
+                        {
+                            await process.WaitForExitAsync(timeoutCts.Token).ConfigureAwait(false);
+                        }
+                        catch (OperationCanceledException) when (timeoutCts.Token.IsCancellationRequested && !cancellationToken.IsCancellationRequested)
+                        {
+                            Logger.LogError("Process timed out after {Timeout}ms: {Command}",
+                                timeout.Value.TotalMilliseconds, command);
+
+                            try
+                            {
+                                process.Kill(entireProcessTree: true);
+                            }
+                            catch (Exception killEx)
+                            {
+                                Logger.LogWarning(killEx, "Failed to kill timed-out process");
+                            }
+
+                            return (false, string.Empty, $"Process timed out after {timeout.Value.TotalMilliseconds}ms");
+                        }
+                    }
+                    else
+                    {
+                        await process.WaitForExitAsync(cancellationToken).ConfigureAwait(false);
+                    }
+                }
+                catch (Win32Exception ex) when (ex.NativeErrorCode == 2)
+                {
+                    Logger.LogError("Command not found: {Command}", command);
+                    return (false, string.Empty, "Command not found");
+                }
+                catch (Win32Exception ex)
+                {
+                    Logger.LogError(ex, "Win32 error starting process: {Command}", command);
+                    return (false, string.Empty, ex.Message);
+                }
+                catch (InvalidOperationException ex)
+                {
+                    Logger.LogError(ex, "Invalid operation starting process: {Command}", command);
+                    return (false, string.Empty, ex.Message);
+                }
+
+                string output, error;
+                lock (outputLock)
```

Comment:
Good threading awareness.   We should take a deeper look at the lock strategies and such here when we do performance tuning.